extmod/vfs_posix_file.c

```diff
@@ -96,14 +97,13 @@ mp_obj_t mp_vfs_posix_file_open(const mp_obj_type_t *type, mp_obj_t file_in, mp_
 
     mp_obj_t fid = file_in;
 
+    int fd;
```

Comment:
I don't think this line or any below it needs to change.  The only change should be `o->fd=-1` just after creating the object.

Reply:
Yeah ok, the other changes could go in a separate commit (/pr); they're just a code size improvement with no functional change so I left them in.

Reply:
Are they size improvements?  I would think the compiler is pretty good at optimising the `return`.

Reply:
Oh, I was sure I saw a size improvement, but sure enough I just compiled Unix before and after this change with no difference in size.
Maybe I had previously looked at size before / after a rebase as well.

Regardless, I've simplified this PR as suggested!